### PR TITLE
TSM tests are all passing!

### LINKED ISSUE #20: Move TSM calculation tests to the `pytest` directory -> get them passing
This can be closed once we are confident in the calculations being done by the refactored TSM model.
The testing code may need to be updated to work with the refactored pattern.

**Tasks:**

- [x] Right now `xarray.apply_ufunc` is not working with Reynold Number calculation (which contains if/else). This needs to be resolved before this issue can be solved. -> solved w/ #39 
- [x] Double check that the `TSM Manual Checks_1D timestep.xlsx` constant / input values match our defaults. Even if they do, since defaults might change, we should hard code this into the test suite.
- [ ] Create a test for every case in the XLSX file, instantiate a fresh model and run each time.
- [ ] Assess if there are calculation issues, if so, try to diagnose them.

### PR DESCRIPTION
**Pull request by xaviernogueira**
Closes #20 

**Changes:**
* Changes to TSM process functions to align with our QA/QC Excel sheet / hec-ras.
* Fully passing test suite at a BLANK tolerance level - WIP
* A handy script for debugging.

**WIP - Next Steps:**

- [x] Fix our answers to account for the scale factor applied in the QA/QC Excel provided by @imscw95 .
- [ ] See how low we can get the tolerance factor and still pass all tests.

**Discussion of an alternative approach:** @aufdenkampe we talked about solving all tests at once using an extra `xarray.Dataset` dimension. While in a notebook environment, I would agree this is the best way to verify calculations, in `pytest` you would need to make a fixture of the master `xarray.Dataset`, alter along using hardcoded values, then read slices of that into `EnergyBudget(hotstart_dataset=dataset)` which is the non-default pattern. Then in unique functions for each check, you would have your assert statements, as in `pytest` you want unique functions per test for helpful feedback and to avoid early exceptions. I decided to not pursue this because:
1. I already had a strong approach working, and I was looking to triage my time.
2. But mainly it separates the altered inputs from the answers, which makes things more confusing (I.e., scrolling up and down). A solution here is to include an assert statement that checks that the input value is changed, which allows the user to infer that it the *only* value that is altered. But I prefer the straightforward approach of just a fresh init for each test, no inference required!

### CODE REVIEW

**Review by aufdenkampe — APPROVED** [review]
@xaviernogueira, this looks good.

Thanks for your clear explanation of the the two approaches considered to model testing (i.e. comparing to RAS-1D-WQ outputs), and your reasons for selecting the pattern that aligns with your existing pytests. That makes sense to me.

**Comment by xaviernogueira** [comment]
@aufdenkampe @imscw95 

**Update:**
All tests pass down to a tolerance of +/- 0.0000001 except three:
```
======================================= short test summary info =======================================
FAILED tests/test_5_tsm_calculations.py::test_changed_water_temp_c - assert 39.999152040147095 ± 4.0e-06 == 39.99618297
FAILED tests/test_5_tsm_calculations.py::test_changed_air_temp_c - assert 19.999991244551833 ± 2.0e-06 == 19.99999407
FAILED tests/test_5_tsm_calculations.py::test_changed_wind_b - assert 19.999980923751853 ± 2.0e-06 == 19.99995768
==================================== 3 failed, 12 passed in 9.16s =====================================
```

As you can see, none fail that bad. I am going to investigate where the math isn't adding up, but things are looking good.

**Still to do:**

- [ ] Upload the Excel to the repo?
- [ ] Change the `mf_cp_water` function in TSM to numpy select so we can use `numba`.
- [ ] Debug the remaining three failed tests.

File: src/clearwater_modules/tsm/processes.py [hunk 1]
```diff
@@ -440,28 +440,27 @@ def mf_density_air_sat(water_temp_k: xr.DataArray, esat_mb: float, pressure_mb:
...
+    # )
+
+    return xr.where(
+        water_temp_c <= 0.0, 4218.0,
+            xr.where(water_temp_c <= 5.0, 4202.0,
+                xr.where(water_temp_c <= 10.0, 4192.0,
+                    xr.where(water_temp_c <= 15.0, 4186.0,
+                        xr.where(water_temp_c <= 20.0, 4182.0,
+                                xr.where(water_temp_c <= 25.0, 4180.0, 4178.0
+        ))))))
```
[the remark this answers is not in the document]
Comment (xaviernogueira) [reply src/clearwater_modules/tsm/processes.py]: Done

**Comment by imscw95** [comment]
Does it work when you set wind_a = .3 and wind_b = 1.5? I believe these values should be on the order of 10^-7 and 10^-6 respectively... so the parameter entry should be in the above suggested range if the code includes the divide by 1000000, or should be entered as 3*10^-7 and 1.5*10^-6.

**Comment by xaviernogueira** [comment]
@imscw95 yes it works fine normally it just doesn't get the right answer when it wind_b changes.

**Comment by xaviernogueira** [comment]
@imscw95 

So I think I figured it out. In the excel there was no division factor, and I realized that you changed multiple wind variables in the "Change win_b" columns when compared to the inputs on the far left.

I simply changed the test to switch wind_b with 1.0, and opposed to 1.0e-6. All tests pass

**Pull request merged**